rules-reviewed/openshift/socket-communication.windup.xml

```diff
@@ -0,0 +1,148 @@
+                    In an OpenShift environment, it is required to use Http(s)-based communication.  
+                  </message>
+                  <tag>socket</tag>
+                </hint>
+              </iteration>
+            </perform>
+            <where param="niochannelclass">
+              <matches pattern="(NetworkChannel|MulticastChannel|DatagramChannel|AsynchronousSocketChannel|SocketChannel)"/>
+            </where>
+        </rule>
+        <rule id="socket-communication-00003">
+            <when>
+               <javaclass references="java.net.{socketclass}{*}">
+                   <location>METHOD_CALL</location>
+               </javaclass>
+            </when>
+            <perform>
+              <iteration>
+                <hint title="Java APIs for socket communication are used" effort="3" category-id="cloud-optional">
+                  <message>
+                    Java APIs for socket communication are used in the file.  
+                    Recommendation: avoid direct use of client sockets. Consider using Java EE standard communication features, especially loosely coupled protocols (JAX-WS, JAX-RS) instead.  
+                    In an OpenShift environment, it is required to use Http(s)-based communication.  
+                  </message>
+                </hint>
+              </iteration>
+            </perform>
+            <where param="socketclass">
+              <matches pattern="(Socket|MulticastSocket|DatagramSocket|InetSocketAddress)"/>
+            </where>
+        </rule>
+        <rule id="socket-communication-00004">
+            <when>
+                <or>
+                  <javaclass references="java.net.ServerSocket{*}">
+                      <location>CONSTRUCTOR_CALL</location>
+                      <location>IMPORT</location>
+                      <location>INHERITANCE</location>
+                      <location>METHOD_CALL</location>
+                      <location>VARIABLE_DECLARATION</location>
+                  </javaclass>
+                    <javaclass references="java.nio.channels.AsynchronousServerSocketChannel{*}">
+                        <location>CONSTRUCTOR_CALL</location>
+                        <location>IMPORT</location>
+                        <location>VARIABLE_DECLARATION</location>
+                    </javaclass>
+                    <javaclass references="java.nio.channels.ServerSocketChannel{*}">
+                        <location>CONSTRUCTOR_CALL</location>
+                        <location>IMPORT</location>
+                        <location>VARIABLE_DECLARATION</location>
+                    </javaclass>
+                </or>
+            </when>
+            <perform>
+                <hint title="Java APIs for server sockets are used" effort="3" category-id="cloud-mandatory">
```

Comment:
be careful also for duplicated titles. we should take care and try to have unique titles